### PR TITLE
WIP: rework filelocking used for scc sync

### PR DESCRIPTION
**Pull request by lucidd**
## What does this PR change?

This PR reworks the file locking we use for synchronize scc refresh between tomcat and taskomatic in an effort to simplify the code and usage. There also have been some issues indicating that the current locking may not always work but since this issue has not been reproduced so far its unclear if the new code resolves this.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
- No tests

- [ ] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/common/util/FileLocks.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2018 SUSE LLC
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/common/util/FileLocks.java:2]: old copyright year

File: java/code/src/com/redhat/rhn/common/util/FileLocks.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.attribute.UserPrincipal;
+import java.nio.file.attribute.UserPrincipalLookupService;
+import java.util.function.Supplier;
+
+/**
+ * SCCRefreshLock - locks the SCC refresh process
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/common/util/FileLocks.java:34]: old comment from former class name

File: java/code/src/com/redhat/rhn/common/util/FileLocks.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+ */
+public class FileLocks {
+
+    // Logger instance
+    private static Logger log = Logger.getLogger(FileLocks.class);
+
+    public static FileLocks SCC_REFRESH_LOCK =
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/common/util/FileLocks.java:41]: As it is public, does it need a comment?

**Review by mcalmer — CHANGES_REQUESTED** [review]
Just small issues in doc and checkstyle is also not green. But the code looks good.
Not sure if there was a bug to name it in the changelog. If not you can also click the "no changelog" checkbox

**Comment by lucidd** [comment]
> Not sure if there was a bug to name it in the changelog. If not you can also click the "no changelog" checkbox

There is potentially a bug related to this but its not confirmed yet that the locking is the issue and if this PR fixes it. I will leave this PR open until we have more information.

**Review by mcalmer — APPROVED** [review]

**Pull request merged**